tests/cases/conformance/types/primitives/string/assignFromStringInterface2.ts

```diff
@@ -11,8 +11,8 @@ interface NotString {
     indexOf(searchString: string, position?: number): number;
     lastIndexOf(searchString: string, position?: number): number;
     localeCompare(that: string): number;
-    match(regexp: string): string[];
-    match(regexp: RegExp): string[];
+    match(regexp: string): RegExpMatchArray;
+    match(regexp: RegExp): RegExpMatchArray;

```

Comment:
why did these change? (and in stringLiteralTypeIsSubtypeOfString). Is it because `RegExpMatchArray` was formerly assignable to `string[]`, but no longer is?

(or maybe the other way round, I never get variance the right way round the first time.)

Reply:
i think it's the other way around, that `string[]` used to be assignable to `RegExpMatchArray` but no longer is because of:
```
tests/cases/conformance/types/primitives/string/assignFromStringInterface2.ts(42,1): error TS2322: Type 'NotString' is not assignable to type 'String'.
  The types returned by 'match(...)' are incompatible between these types.
    Property '0' is missing in type 'string[]' but required in type 'RegExpMatchArray'.
```

i assume that since there were no new errors in DefinitelyTyped that this shouldn't be an issue?